org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                    };
+                    foreach ( KeyValuePair<int, string> kvp in stepsPath )
+                    {
+                        var stepQuery = from s in _context.Steps
+                                       join pa in _context.PersonAliases on s.PersonAliasId equals pa.Id
+                                       where s.StepTypeId == kvp.Key && pa.Id == s.PersonAliasId && pa.PersonId == person.Id
+                                       select s.CompletedDateTime;
+                        var stepResult = stepQuery.FirstOrDefault().ToStringSafe();
+                        stepResult = stepResult != "" ? ConvertDate( ( DateTime ) stepResult.AsDateTime() ) : stepResult;
+                        Debug.WriteLine( "Discpleship Step Path : " + kvp.Value + " | completed date: " + stepResult );
+                        properties.Add( new HubspotPropertyUpdate() { property = kvp.Value, value = stepResult } );
+                    }
+
+                    //// First time Next Step Class Scheduled
+                    //var nscQuery = from h in _context.Histories
+                    //               where h.ValueName == "Next Step Class Scheduled" && h.EntityId == person.Id
+                    //               orderby h.Id ascending
+                    //               select h.NewValue;
+                    //var nscResult = nscQuery.FirstOrDefault().ToStringSafe();
+                    //properties.Add( new HubspotPropertyUpdate() { property = "next_steps_class_registration", value = nscResult } );
+                    //Debug.WriteLine( "First Time Next Step Class Scheduled : " + nscResult );
+
+
+                    try
+                    {
+                        //Update the Contact in Hubspot
+                        MakeRequest(current_id, url, properties, 0);
+                        //WriteToLog( string.Format( "    After Request: {0}", watch.ElapsedMilliseconds ) );
+                    }
+                    catch ( Exception err )
+                    {
+                        ExceptionLogService.LogException(new Exception($"Hubspot Sync Error{Environment.NewLine}{err}{Environment.NewLine}Current Id: {current_id}{Environment.NewLine}Exception from Job:{Environment.NewLine}{err.Message}{Environment.NewLine} - Record Count:{i}"));
+                    }
+                }
+                //WriteToLog( string.Format( "    End of iteration: {0}", watch.ElapsedMilliseconds ) );
+                //watch.Stop();
+                WriteToLog( string.Format( i + " of " + contacts.Count() + " Contacts Patched." ) );
+                WriteToLog( string.Format( "Last Person: " + person.FullName + "( " + person.Id + " )" ) );
+
+                // Output Job Stats
+                var resultMsg = new StringBuilder();
+                resultMsg.AppendFormat( i + " of " + contacts.Count() + " Contacts Patched." );
+                resultMsg.AppendFormat( "\n Last Person: " + person.FullName + "(" + person.Id + ")" );
```

Comment:
First, this seems like a very simple string. Why are we using `StringBuilder` here? We could use string interpolation (`$"Some text { someVariable }"`) and make this a lot more readable.

Second, why does it matter what the last person processed was? Are we going to use this info for anything? If not, can we remove it?